cmd/neofs-node/grpc.go

```diff
@@ -42,6 +43,10 @@ func initGRPC(c *cfg) {
 			return
 		}
 
+		if connLimit := sc.ConnectionLimit(); connLimit > 0 {
+			lis = netutil.LimitListener(lis, connLimit)
```

Comment:
this opens up vulnerability to idle clients: when `connLimit` of them become idle and dont close the connection, all new conns are refused. Overall, i'd not delegate this job to some hard-limiting tool like `netutil.LimitListener`. There are some suggestions in the issue https://github.com/nspcc-dev/neofs-node/issues/1240#issuecomment-1084216928

Reply:
we can control it with server params: https://pkg.go.dev/google.golang.org/grpc#KeepaliveParams

> all new conns are refused

not refused but blocked until some connection is closed in fact

Reply:
> we can

i know we can, but we dont

> not refused but blocked

to be more precise, yeah, but the fact remains - no connection

Reply:
> i know we can, but we don't

we can start with this PR, not sure what bound values to choose. depends on whether you still think this approach is usable or not
@roman-khimov

Reply:
> to be more precise, yeah, but the fact remains - no connection

Just like with any other limit. Unlimited connections is a vulnerability of one kind, limited is a vulnerability of the other.

The proposal is to add it this way for now and then if/when needed try tuning it. If there are better options found, we'll implement them. This handle itself doesn't cost a lot.

Reply:
ok lets see how it'll go